File: 686-repeated-string-match/686-repeated-string-match.py
class Solution:
    def repeatedStringMatch(self, a: str, b: str) -> int:
        
        count = 0
        s = ""
        while (len(s) < len(b)):
            s += a
            count += 1
        
        if b in s:
            return count
        elif b in s + a:
            return count + 1
        else:
            return -1
    

File: 686-repeated-string-match/686-repeated-string-match.cpp
class Solution {
public:
    int repeatedStringMatch(string a, string b) 
    {
        string s = "";
        int count = 0;
        while (s.length() < b.length())
        {
            s += a;
            count++;
        }
        
        if (s.find(b) != -1)
            return count;
        
        s += a;
        if (s.find(b) != -1)
            return count + 1;
        
        return -1;
    }
};
